samples/performance/constant_data/constant_data_tutorial.md

```diff
@@ -0,0 +1,399 @@
+
+## **Dynamic Descriptor Sets**
+
+### **Introduction**
+
+Dynamic descriptor sets differ to the regular descriptor sets because they allow an offset to be specified when we aere *binding* (`vkCmdBindDescriptorSets`) the descriptor set. This dynamic offset can be used in addition to the base offset used at the time of updating the descriptor set. 
+
+One case in which this can be useful is:
+
+1. Allocating one giant **uniform buffer object** containing all the world matrices of the meshes in your scene. 
+2. Allocating a **descriptor set** with a binding containing the `VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER_DYNAMIC` flag that then points to the **UBO** you just created.
+3. In our draw call, for each mesh, *dynamically* offset into the giant **uniform buffer object**.
+
+### **Performance**
+
+![dynamic ubo](images/dynamic_descriptor_set_performance.jpg)
+
+In the screenshot above we can see the load/store cycles stay roughly around the same compared to [static uniform buffer objects](#uniform-buffer-objects). However, the frame time goes up from 16.7 ms to 20.9 ms. This is due to the extra time you need to spend every frame determining the dynamic offsets, that you need to send in the bind call (`vkCmdBindDescriptorSets`).
+
+## **Update-after-bind Descriptor Sets**
+
+### **Introduction**
+
+Traditionally, **descriptor sets** require updating before they are bound to a command buffer - any further updates after it is bound will invalidate the command buffer it is bound to. However, this can be considered an "overly cautious" restriction when we realise that the command buffer isn't actually executed until it's submitted on a queue. This is where newer versions of Vulkan have introduced the concept of "update after bind". 
+
+Essentially it adds in a binding flag to descriptor set layouts which allows the contents of the **descriptor set** to be updated up until the command buffer *is submitted to the queue*, rather than when the descriptor set is *bound to the command buffer*.
+
+***Note:** Update-after-bind bindings cannot be used with dynamic descriptor sets.*
+
+### **Performance**
+
+This should come with zero performance costs, and as a result this method is designed purely for offering flexibility to your codebase. 
+
+## **Buffer Object Arrays**
+
+### **Introduction**
+
+Another approach, which can be likened to a dynamic descriptor set, is a buffer object array. This is the concept of allocating all of your constant data *upfront* in a large buffer, and writing the entire buffer to a descriptor set. This means in any one shader invocation we have access to all of the model data for the entire scene, at the benefit of only needing to bind one descriptor set per entire draw call.
+
+You can use either a `uniform` or a `buffer` storage type in your shader code to achieve this. However, since `uniform`s can only carry 16 KB of data, this tutorial will use a `buffer`. 
```

Comment:
Arm specific limit. In general wouldn't it be better to say that if the shader is only reading data and the buffer is smaller than the UBO size limit, a UBO would be better. When the shader writes data, use a storage buffer?